e2e/test/scenarios/question/offset.cy.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import type { StructuredQuestionDetails } from "e2e/support/helpers";
+import {
+  createQuestion,
+  enterCustomColumnDetails,
+  getNotebookStep,
+  openNotebook,
+  popover,
+  restore,
+} from "e2e/support/helpers";
+
+const { ORDERS_ID } = SAMPLE_DATABASE;
+
+describe("scenarios > question > offset", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should allow using OFFSET as a CASE argument (metabase#42377)", () => {
+    const formula = "Sum(case([Total] > 0, Offset([Total], -1)))";
+    const name = "Aggregation";
+    const questionDetails: StructuredQuestionDetails = {
+      query: {
+        "source-table": ORDERS_ID,
+        limit: 5,
+      },
+    };
+    createQuestion(questionDetails, { visitQuestion: true });
+    openNotebook();
+
+    cy.icon("sum").click();
+    getNotebookStep("summarize")
+      .findByText("Pick the metric you want to see")
+      .click();
+    popover().contains("Custom Expression").click();
+    enterCustomColumnDetails({ formula, name });
+
+    cy.on("uncaught:exception", error => {
+      expect(error.message.includes("Error normalizing")).not.to.be.true;
```

Comment:
what kind of error do we expect? is there any other way to verify we don't have normalization error?

Reply:
> what kind of error do we expect?

We expect no error.

`cy.on` will however catch all the other errors from console, e.g. warnings about missing props coming from React, etc.

> is there any other way to verify we don't have normalization error?

We could wrap `Lib.diagnoseExpression` call in `diagnostics.ts` in a `try`/`catch` block and then the error would be shown to the user. I'm not sure why we haven't done that originally.